samples/snippets/csharp/tutorials/inheritance/base-and-derived.cs

```diff
@@ -0,0 +1,189 @@
+﻿
+// <Snippet1>
+using System;
+
+public enum PublicationType { Misc, Book, Magazine, Article };
+
+public abstract class Publication
+{
+   private bool published = false;
+   private DateTime datePublished;
+   private string pubName;
+   private string pubTitle;
+   private PublicationType pubType;
+   private string copyrName;
+   private int copyrDate;
+   private int totalPages; 
+
+   public Publication(string title, string publisher, PublicationType type)
+   {
+      if (publisher == null)
+         throw new ArgumentNullException("The publisher cannot be null.");
+      else if (String.IsNullOrWhiteSpace(publisher))
+         throw new ArgumentException("The publisher cannot consist only of whitespace.");
+      pubName = publisher;
+  
+      if (title == null)
+         throw new ArgumentNullException("The title cannot be null.");
+      else if (String.IsNullOrWhiteSpace(title))
+         throw new ArgumentException("The title cannot consist only of whitespace.");
+      pubTitle = title;
+
+      pubType = type;
+   }
+
+   public string Publisher
```

Comment:
These properties would be much simpler as read only auto implemented properties.